control-plane/agents/src/bin/core/main.rs

```diff
@@ -88,13 +88,42 @@ pub(crate) struct CliArgs {
     /// If `None` do not limit the number of rebuilds.
     #[clap(long)]
     max_rebuilds: Option<NumRebuilds>,
+
+    #[clap(flatten)]
+    thin_args: ThinArgs,
 }
 impl CliArgs {
     fn args() -> Self {
         CliArgs::parse()
     }
 }
 
+/// Cluster wide thin provisioning parameters.
+#[derive(Debug, clap::Parser, Clone)]
+pub(crate) struct ThinArgs {
+    /// The allowed pool commitment limit when dealing with thin provisioned volumes.
+    /// Example: If the commitment is 250 and the pool is 10GiB we can overcommit the pool
+    /// up to 25GiB (create 2 10GiB and 1 5GiB volume) but no further.
+    #[clap(long, env = "POOL_COMMITMENT_%", value_parser = value_parse_percent, default_value = "250%")]
+    pool_commitment: u64,
+    /// When creating pool replicas for a new volume, each replica pool must have at least this
+    /// much free space percentage of the volume size.
+    /// Example: if this value is 40 and the pool has 40GiB free, then the max volume size allowed
+    /// to be created on the pool is 100GiB.
+    #[clap(long, env = "VOLUME_COMMITMENT_%_INITIAL", value_parser = value_parse_percent, default_value = "40%")]
+    volume_commitment_initial: u64,
+    /// When creating pool replicas for an existing volume, each replica pool must have at least
+    /// this much free space percentage of the volume size.
+    /// Example: if this value is 40, the pool has 40GiB free, then the max volume size allowed
+    /// to be created on the pool is 100GiB.
```

Comment:
The doc comment still look identical? Should we mention how one differs from other?

Reply:
One is for new, one is not existing volumes, that's the difference.
Yes that's a good idea, I'll say how they differ.